File: src/subside/plugins/koth/commands/CommandEnd.java
package subside.plugins.koth.commands;

import org.bukkit.command.CommandSender;

import subside.plugins.koth.commands.CommandHandler.CommandCategory;
import subside.plugins.koth.exceptions.CommandMessageException;
import subside.plugins.koth.gamemodes.RunningKoth.EndReason;
import subside.plugins.koth.modules.Lang;
import subside.plugins.koth.utils.IPerm;
import subside.plugins.koth.utils.MessageBuilder;
import subside.plugins.koth.utils.Perm;

public class CommandEnd extends AbstractCommand {

    public CommandEnd(CommandCategory category) {
        super(category);
    }

    @Override
    public void run(CommandSender sender, String[] args) {
        if (args.length > 0) {
            getPlugin().getKothHandler().endKoth(args[0], EndReason.GRACEFUL);
            throw new CommandMessageException(new MessageBuilder(Lang.COMMAND_TERMINATE_SPECIFIC_KOTH).koth(getPlugin().getKothHandler(), args[0]));
        } else {
            getPlugin().getKothHandler().endAllKoths(EndReason.GRACEFUL);
            throw new CommandMessageException(Lang.COMMAND_TERMINATE_ALL_KOTHS);
        }
    }

    @Override
    public IPerm getPermission() {
        return Perm.Control.END;
    }

    @Override
    public String[] getCommands() {
        return new String[]{"end"};
    }
    
    @Override
    public String getUsage() {
        return "/koth end [koth]";
    }

    @Override
    public String getDescription() {
        return "Gracefully ends a (specific) koth";
    }

}

File: src/subside/plugins/koth/commands/control/CommandEnd.kt
package subside.plugins.koth.commands.control

import org.bukkit.command.CommandSender
import subside.plugins.koth.KothManager
import subside.plugins.koth.commands.Command
import subside.plugins.koth.gamemodes.RunningKoth
import subside.plugins.koth.modules.Lang
import subside.plugins.koth.utils.IPerm
import subside.plugins.koth.utils.MessageBuilder
import subside.plugins.koth.utils.Perm

class CommandEnd : Command {
    override fun run(kothManager: KothManager, sender: CommandSender, args: Array<String>) {
        if (args.isEmpty()) {
            kothManager.kothHandler.endAllKoths(RunningKoth.EndReason.GRACEFUL)
            MessageBuilder(Lang.COMMAND_TERMINATE_ALL_KOTHS).send(sender)
        } else {
            kothManager.kothHandler.endKoth(args[0], RunningKoth.EndReason.GRACEFUL)
            MessageBuilder(Lang.COMMAND_TERMINATE_SPECIFIC_KOTH)
                    .koth(kothManager.kothHandler, args[0])
                    .send(sender)
        }
    }

    override val permission: IPerm = Perm.Control.END
    override val commands: Array<String> = arrayOf("end")
    override val usage: String = "/koth end [koth]"
    override val description: String = "Gracefully ends a (specific) KoTH"
}